src/components/MapPlot.jsx

```diff
@@ -61,8 +62,28 @@ export const MapPlot = React.forwardRef((props, ref) => {
   const onDefinedPoint = ({ lngLat }) => {
     dispatch(actions.addDefinedPoint(lngLat));
   };
-
+  const layers = [
```

Comment:
does it make sense to wrap this in `useMemo`?

Reply:
Using useMemo in this context might not be the most appropriate choice. useMemo is primarily used for memoizing the result of a computation to avoid expensive recalculations when dependencies haven't changed. 

Instead, you should consider using useCallback if you want to memoize the function itself to prevent unnecessary re-creations of the function on every render.

`const onDefinedPoint = useCallback(({ lngLat }) => {
    dispatch(actions.addDefinedPoint(lngLat));
  }, [dispatch]);`

Any Suggestion ?

Reply:
Ah sorry I was talking about `layers` not onDefinedPoint

Reply:
Sorry, my bad. 

Sure we cn use useMemo. By using useMemo for the layers array, it ensures that the PathLayer and ScatterplotLayer instances are only created when plotPaths or plotPoints change, which can help improve the performance of our component.

`const layers = useMemo(() => [
    new PathLayer({
      id: "path-layer",
      data: plotPaths,
      getPath: d => d.path,
      getColor: d => d.color,
      pickable: true,
      widthMinPixels: 4,
      widthMaxPixels: 8
    }),
    new ScatterplotLayer({
      id: "scatter-layer",
      data: plotPoints,
      pickable: true,
      opacity: 0.8,
      getFillColor: p => p.color,
      radiusMinPixels: 6,
      radiusMaxPixels: 8
    })
  ], [plotPaths, plotPoints]);`
  
  Unfortunatly as i am out of station , i cannot test the change. Can you please test with this change and let me know if it works fine.